code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/source_model_validation/ContainerBaseSourceModelValidator.xtend

```diff
@@ -0,0 +1,59 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.source_model_validation
+
+import de.fhdo.lemma.model_processing.annotations.Before
+import de.fhdo.lemma.model_processing.annotations.SourceModelValidator
+import de.fhdo.lemma.model_processing.builtin_phases.source_model_validation.AbstractXtextSourceModelValidator
+import de.fhdo.lemma.model_processing.builtin_phases.source_model_validation.SourceModelValidationMode
+import de.fhdo.lemma.model_processing.languages.XtextImportUriConverterKt
+import de.fhdo.lemma.operation.OperationModel
+import de.fhdo.lemma.operation.OperationPackage
+import org.eclipse.xtext.resource.XtextResource
+import de.fhdo.lemma.operation.Container
+import org.eclipse.xtext.validation.Check
+
+/**
+ * The container base model validator is responsible for checking general aspects for
+ * the container base code generator.
+ *
+ * @author Philip Wizenty(philip.wizenty@fh-dortmund.de)
+ */
+@SourceModelValidator(supportedFileExtensions='operation',
+    validationMode=SourceModelValidationMode.XTEXT
+)
+class ContainerBaseSourceModelValidator extends AbstractXtextSourceModelValidator {
+
+    /*
+     * Get namespace for the operation model package
+     */
+    override getLanguageNamespace() {
+        return OperationPackage.eNS_URI
+    }
+
+    /*
+     * Prepare the source model for the testing process
+     */
+    @Before
+    def prepareSourceModel(XtextResource xtextResource) {
+        if (xtextResource.getContents().isEmpty())
+            return;
+        XtextImportUriConverterKt.convertToAbsoluteFileUrisInPlace((xtextResource.getContents()
+            .get(0) as OperationModel
+        ).imports, xtextResource)
+    }
+
+    /*
+     * Check if the source model uses the container base technology
+     */
+    @Check
+    def checkContainer(Container container) {
+        val technology = container.technologies.findFirst[t |
```

Comment:
A much better approach would be to use Xtend's `exists` function if you do not need the `technology` variable at all.